### ISSUE TITLE
Review cursor: don't report "top"/"bottom" message when moving on the first/last line of the current object

### ISSUE THREAD

**Issue by Andre9642** [nvaccess/nvda#9551]
### Steps to reproduce:
Press NVDA+Control+home / NVDA+Control+end in laptop layout. Or shift+numpad7 / shift+numpad9 in desktop layout.

### Actual behavior:
NVDA reports (through speech and braille) "top" or "bottom" flash message, then reports the line.

### Expected behavior:
NVDA should just reports the line.

Note: I work with command line a lot and I often use these commands. In my opinion, these flash messages are painful, especially in braille. I'd suggest to remove them.

#### NVDA installed/portable/running from source:
Any

#### NVDA version:
Any

#### Windows version:
N/A

#### Name and version of other software in use when reproducing the issue:
N/A

#### Other information about your system:
N/A

#### Does the issue still occur after restarting your PC?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes, same behavior.

**Comment by feerrenrut** [comment]
Please outline actual steps to reproduce, eg which application which control what text. This helps to ensure everyone is talking about the same thing.

`def script_review_previousLine` also announces "top" when the top line is reached. It helps to have this behaviour to differentiate between multiple equal lines, and reaching the limit of the object.

I suspect there are a few reasons for this behaviour:
- help to orient users after a change in position.
- to have symmetrical behaviour with `script_review_previousLine`

Before we can make a change here, we should get more information from the community.

**Comment by feerrenrut** [comment]
I have had a chat to @michaelDCurran about this. 

We have agreed that it is ok to remove "top" and "bottom" from `script_review_top` and `script_review_bottom`. However, to keep these the behaviour symmetrical with the left / right commands, then "left" and "right" should be removed from `script_review_startOfLine` and `script_review_endOfLine`

We realised that `script_review_previousLine` requires "top" to be announced when the script is triggered but the position does not change, in order to be able to differentiate encountering repeated lines from hitting the top boundary. This is not the case for `script_review_top` since it will always move to the top, and the behaviour is not different on subsequent calls. 

Still, please update the description on this issue to explain the details more clearly.

**Comment by fisher729** [comment]
Hi.

Pardon my confusion, but I still see this being reported. As I type this, if I press Numpad 3, I hear right, followed by blank. This issue, therefore, needs further explanation, especially if this is still no longer intended. I'm using NVDA 2020.2beta2.

**Comment by michaelDCurran** [comment]
This issue was specific to the move to top / move to bottom scripts for review cursor. These no longer redundantly announce "top" and "bottom". The messages are not required in these scenarios as it is obvious to the user that they hit the top / bottom because they deliberately went there.  However, the other review scripts (previs and next line/word/char) still say these messages if you try to move past the edge of the text. If users also want the messages removed in that those scenarios, a new issue should be opened.

**Comment by fisher729** [comment]
Hi.

I see that now, and fully understand what is being solved here. However,
the wording in the What's New does not match what is being said here, and
requires a more specific explanation. Perhaps this can work to replace both
entries:
•NVDA no longer reports "top", "bottom", "left" or "right"  messages when
an input gesture is used to directly move the review cursor to the first or
last line or character of the current navigator object. (#9551)

**Comment by michaelDCurran** [comment]
I agree. Thanks for the corrected wording here.
